src/main/java/de/tum/in/www1/artemis/domain/ComplaintResponse.java

```diff
@@ -8,31 +11,69 @@
 import org.hibernate.annotations.Cache;
 import org.hibernate.annotations.CacheConcurrencyStrategy;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
  * A ComplaintResponse.
+ *
+ *  The createdDate in {@link AbstractAuditingEntity#createdDate} has a special meaning in this entity as it is used to calculate the lock status. See also
+ *  {@link ComplaintResponse#isCurrentlyLocked()}
  */
 @Entity
 @Table(name = "complaint_response")
 @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public class ComplaintResponse extends DomainObject {
+public class ComplaintResponse extends AbstractAuditingEntity {
 
     @Column(name = "response_text")
     @Size(max = 2000)
     private String responseText;
 
+    /**
+     * will be set as soon as the reviewer has submitted his complaint evaluation
+     */
     @Column(name = "submitted_time")
     private ZonedDateTime submittedTime;
 
     @OneToOne
     @JoinColumn(unique = true)
+    @JsonIgnoreProperties(value = "complaintResponse", allowSetters = true)
     private Complaint complaint;
 
     @ManyToOne
     private User reviewer;
 
+    /**
+     * Calculates if the complaint response is still time locked or not
+     * @return true if complaint response is time locked, false otherwise
+     */
+    @JsonProperty("isCurrentlyLocked")
+    public boolean isCurrentlyLocked() {
+        if (getCreatedDate() == null) {
+            return false;
+        }
+        ZonedDateTime createdDateInUTC = ZonedDateTime.ofInstant(getCreatedDate(), ZoneOffset.UTC);
+
+        ZonedDateTime lockedUntil = createdDateInUTC.plusMinutes(COMPLAINT_LOCK_DURATION_IN_MINUTES);
+
+        return ZonedDateTime.now(ZoneOffset.UTC).isBefore(lockedUntil);
```

Comment:
I am just curious, it this `ZonedOffset.UTC` needed here since the date is calculated on server side which operates in UTC anyway?

Reply:
Not need but doesnt hurt